### PR TITLE
WIP: Add functionality to support request type

### PR DESCRIPTION
**Pull request by simonteozw**
Hey guys you all can take a look at leave some suggestions, it mostly works I am wondering but renaming some of the files (eg. AddCommand -> AddContactCommand / AddPersonCommand) to make things clearer. Need some input haha

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java [hunk 1]
```diff
@@ -25,7 +27,8 @@
     /**
      * Used for initial separation of command word and args.
      */
-    private static final Pattern BASIC_COMMAND_FORMAT = Pattern.compile("(?<commandWord>\\S+)(?<arguments>.*)");
+    private static final Pattern BASIC_COMMAND_FORMAT = Pattern.compile("(?<type>[A-Z]\\s+)?"
```
Comment (UncleGrandpa925) [inline src/main/java/seedu/address/logic/parser/AddressBookParser.java:30]: I think you pattern did not match our command. Here is my suggestion: 
`(?<type>[A-Z]\s+)?(?<commandWord>\S+)(?<arguments>.*)`

File: src/main/java/seedu/address/model/person/Person.java [hunk 2]
```diff
@@ -14,6 +14,7 @@
  * Guarantees: details are present and not null, field values are validated, immutable.
  */
 public class Person {
```
Comment (UncleGrandpa925) [inline src/main/java/seedu/address/model/person/Person.java]: Maybe we rename this class to Contact now?

**Review by UncleGrandpa925 — CHANGES_REQUESTED** [review]
Nice code overall!

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java (hunk 1 above)
Comment (simonteozw) [reply src/main/java/seedu/address/logic/parser/AddressBookParser.java:30]: Ok I tried to change to this but err java gives me an error: Illegal escape character in String literal

Comment (UncleGrandpa925) [reply src/main/java/seedu/address/logic/parser/AddressBookParser.java:30]: Ahh I see, just add another \ before each of the current \

**Review by UncleGrandpa925 — APPROVED** [review]
Okay let's merge it

**Pull request merged**